src/gui/FileBrowser.cpp

```diff
@@ -162,17 +196,52 @@ bool FileBrowser::filterItems( const QString & filter, QTreeWidgetItem * item )
 }
 
 
+void filterDirectories(QString& dirstring, const QString& filterstring, bool checked)
+{
+	if (!checked)
+	{
+		dirstring = dirstring.replace(filterstring, "");
+	}
+	else
+	{
+		if (!dirstring.contains(filterstring))
+		{
+			dirstring += "*" + filterstring;
+		}
+	}
+	dirstring = dirstring.replace("**", "*");
+	if ((dirstring.length() > 0) && (*dirstring.begin() == '*'))
+	{
+		dirstring = dirstring.remove(0, 1);
+	}
+	if ((dirstring.length() > 0) && (*(dirstring.end()-1) == '*'))
+	{
+		dirstring = dirstring.remove(dirstring.length()-1, 1);
+	}
+}
+
 
 void FileBrowser::reloadTree( void )
 {
 	QList<QString> expandedDirs = m_fileBrowserTreeWidget->expandedDirs();
 	const QString text = m_filterEdit->text();
 	m_filterEdit->clear();
 	m_fileBrowserTreeWidget->clear();
-	QStringList paths = m_directories.split( '*' );
-	for( QStringList::iterator it = paths.begin(); it != paths.end(); ++it )
+	if ((m_showUserContent != nullptr) && (m_showFactoryContent != nullptr))
 	{
-		addItems( *it );
+		filterDirectories(m_directories, m_userDir,
+						  m_showUserContent->isChecked());
+		filterDirectories(m_directories, m_factoryDir,
+						  m_showFactoryContent->isChecked());
+	}
+
+	if (m_directories.length() > 0)
```

Comment:
More of a curiosity question, we didn't have this check before, so if `m_directories` was an empty string we would have an empty entry being added right? We didn't have it before because `m_directories` would never be empty or was it a bug?

Reply:
I think the check wasn't there before because `m_directories` would never be empty. If `m_directories` is empty, the files in the folder where lmms was started from are listed. In my opinion, this is not the intended behavior.